cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMakerCluster.java

```diff
@@ -0,0 +1,447 @@
+        this.replicas = DEFAULT_REPLICAS;
+        this.readinessPath = "/";
+        this.readinessTimeout = DEFAULT_HEALTHCHECK_TIMEOUT;
+        this.readinessInitialDelay = DEFAULT_HEALTHCHECK_DELAY;
+        this.livenessPath = "/";
+        this.livenessTimeout = DEFAULT_HEALTHCHECK_TIMEOUT;
+        this.livenessInitialDelay = DEFAULT_HEALTHCHECK_DELAY;
+        this.isMetricsEnabled = DEFAULT_KAFKA_MIRRORMAKER_METRICS_ENABLED;
+
+        this.mountPath = "/var/lib/kafka";
+        this.logAndMetricsConfigVolumeName = "kafka-metrics-and-logging";
+        this.logAndMetricsConfigMountPath = "/opt/kafka/custom-config/";
+    }
+
+    public static String kafkaMirrorMakerClusterName(String cluster) {
+        return cluster + KafkaMirrorMakerCluster.NAME_SUFFIX;
+    }
+
+    public static String serviceName(String cluster) {
+        return kafkaMirrorMakerClusterName(cluster);
+    }
+
+    public static String logAndMetricsConfigName(String cluster) {
+        return cluster + KafkaMirrorMakerCluster.METRICS_AND_LOG_CONFIG_SUFFIX;
+    }
+
+    public static KafkaMirrorMakerCluster fromCrd(KafkaMirrorMaker kafkaMirrorMaker) {
+        KafkaMirrorMakerCluster kafkaMirrorMakerCluster = new KafkaMirrorMakerCluster(kafkaMirrorMaker.getMetadata().getNamespace(),
+                kafkaMirrorMaker.getMetadata().getName(),
+                Labels.fromResource(kafkaMirrorMaker).withKind(kafkaMirrorMaker.getKind()));
+
+        kafkaMirrorMakerCluster.setReplicas(kafkaMirrorMaker.getSpec() != null && kafkaMirrorMaker.getSpec().getReplicas() > 0 ? kafkaMirrorMaker.getSpec().getReplicas() : DEFAULT_REPLICAS);
+
+        if (kafkaMirrorMaker.getSpec() != null) {
+            kafkaMirrorMakerCluster.setWhitelist(kafkaMirrorMaker.getSpec().getWhitelist());
+            kafkaMirrorMakerCluster.setProducer(kafkaMirrorMaker.getSpec().getProducer());
+            kafkaMirrorMakerCluster.setConsumer(kafkaMirrorMaker.getSpec().getConsumer());
+            String image = kafkaMirrorMaker.getSpec().getImage();
+            kafkaMirrorMakerCluster.setImage(image == null ? KafkaMirrorMakerSpec.DEFAULT_IMAGE : image);
+            kafkaMirrorMakerCluster.setLogging(kafkaMirrorMaker.getSpec().getLogging());
+
+            Map<String, Object> metrics = kafkaMirrorMaker.getSpec().getMetrics();
+            if (metrics != null && !metrics.isEmpty()) {
+                kafkaMirrorMakerCluster.setMetricsEnabled(true);
+                kafkaMirrorMakerCluster.setMetricsConfig(metrics.entrySet());
+            }
+        }
+
+        if (kafkaMirrorMaker.getSpec().getConsumer() != null && kafkaMirrorMaker.getSpec().getConsumer().getAuthentication() != null) {
+            if (kafkaMirrorMaker.getSpec().getConsumer().getAuthentication() instanceof KafkaMirrorMakerAuthenticationTls) {
+                KafkaMirrorMakerAuthenticationTls consumerAuth = (KafkaMirrorMakerAuthenticationTls) kafkaMirrorMaker.getSpec().getConsumer().getAuthentication();
```

Comment:
I think you can factor out this bit (up to the end of the else block on line 157) into a method accepts a `KafkaMirrorMakerClient` and which returns `consumerCertificateAndKey`. You should then be able to also replace the similar copy+pasted block for the producer with a call to the new method.